pyvrp/cpp/educate/Route.h

```diff
@@ -111,7 +111,14 @@ bool Route::isFeasible() const
 
 bool Route::hasExcessCapacity() const { return !isLoadFeasible_; }
 
-bool Route::hasTimeWarp() const { return isTimeWarpFeasible_; }
```

Comment:
This actually had a bug: should have been `!isTimeWarpFeasible_`. Fixed with the proposed changes.

Reply:
Nice catch! This makes`Route::isFeasible()` incorrect as well, which is used in prechecks in exchange and 2OPT, so, in the old situation:
* We skip potentially improving moves if there is a timewarp but `deltaCost >= 0`
* We waste some time evaluating non-improving moves. I wonder if it changes performance?
For the second point the bugfix is a guaranteed improvement, but for the first point it is not as it may be better to skip these moves anyway if they rarely yield improvement. Therefore this bug may actually have been a feature, so it may make sense to test this bugfix independently of the other (guaranteed) performance improvements.

Reply:
I think I introduced this bug last month when I started caching the route feasibilities. It wasn't there during Euro/NeurIPS, or DIMACS.

A very significant part of the performance improvement came from fixing the bug (like 80% or so, but I do not have the exact numbers). The old (wrong) behaviour of `Route::isFeasible()` might have helped incidentally, but almost 2x more iterations seems to more than make up for it!